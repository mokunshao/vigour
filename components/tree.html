<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Tree | Vigour</title>
    <meta name="description" content="A UI components library that based on Vue 2.X .">
    
    
    <link rel="preload" href="/vigour/assets/css/0.styles.83d47a9e.css" as="style"><link rel="preload" href="/vigour/assets/js/app.8d4caa4f.js" as="script"><link rel="preload" href="/vigour/assets/js/8.b83c43f8.js" as="script"><link rel="preload" href="/vigour/assets/js/91.5a3e06e9.js" as="script"><link rel="preload" href="/vigour/assets/js/22.1269dd55.js" as="script"><link rel="preload" href="/vigour/assets/js/23.b2be9338.js" as="script"><link rel="prefetch" href="/vigour/assets/js/1.06fa7144.js"><link rel="prefetch" href="/vigour/assets/js/10.05ff5e33.js"><link rel="prefetch" href="/vigour/assets/js/11.06050baa.js"><link rel="prefetch" href="/vigour/assets/js/12.984c3626.js"><link rel="prefetch" href="/vigour/assets/js/13.8713fdec.js"><link rel="prefetch" href="/vigour/assets/js/14.e765b3a1.js"><link rel="prefetch" href="/vigour/assets/js/15.5781bef9.js"><link rel="prefetch" href="/vigour/assets/js/16.b74592a5.js"><link rel="prefetch" href="/vigour/assets/js/17.ac0437d1.js"><link rel="prefetch" href="/vigour/assets/js/18.657247d1.js"><link rel="prefetch" href="/vigour/assets/js/19.cc8d533c.js"><link rel="prefetch" href="/vigour/assets/js/2.07522eb5.js"><link rel="prefetch" href="/vigour/assets/js/20.13939d82.js"><link rel="prefetch" href="/vigour/assets/js/21.36a359d6.js"><link rel="prefetch" href="/vigour/assets/js/24.44bd11e5.js"><link rel="prefetch" href="/vigour/assets/js/25.514fe187.js"><link rel="prefetch" href="/vigour/assets/js/26.029e729d.js"><link rel="prefetch" href="/vigour/assets/js/27.8b1887f2.js"><link rel="prefetch" href="/vigour/assets/js/28.e52e2e6f.js"><link rel="prefetch" href="/vigour/assets/js/29.e65e0d10.js"><link rel="prefetch" href="/vigour/assets/js/3.ad591584.js"><link rel="prefetch" href="/vigour/assets/js/30.0c5ce916.js"><link rel="prefetch" href="/vigour/assets/js/31.9f0d2fb1.js"><link rel="prefetch" href="/vigour/assets/js/32.a1b5b24c.js"><link rel="prefetch" href="/vigour/assets/js/33.44ea3dec.js"><link rel="prefetch" href="/vigour/assets/js/34.c9b79613.js"><link rel="prefetch" href="/vigour/assets/js/35.4fb228c6.js"><link rel="prefetch" href="/vigour/assets/js/36.e3bbe3c8.js"><link rel="prefetch" href="/vigour/assets/js/37.8fb4eba9.js"><link rel="prefetch" href="/vigour/assets/js/38.fdb9b36c.js"><link rel="prefetch" href="/vigour/assets/js/39.2e552d09.js"><link rel="prefetch" href="/vigour/assets/js/4.da83a827.js"><link rel="prefetch" href="/vigour/assets/js/40.0c0d6600.js"><link rel="prefetch" href="/vigour/assets/js/41.a56fa812.js"><link rel="prefetch" href="/vigour/assets/js/42.a5899037.js"><link rel="prefetch" href="/vigour/assets/js/43.145e0d65.js"><link rel="prefetch" href="/vigour/assets/js/44.d231615f.js"><link rel="prefetch" href="/vigour/assets/js/45.cb95032f.js"><link rel="prefetch" href="/vigour/assets/js/46.7a7a3cd2.js"><link rel="prefetch" href="/vigour/assets/js/47.820bbcee.js"><link rel="prefetch" href="/vigour/assets/js/48.153ebb51.js"><link rel="prefetch" href="/vigour/assets/js/49.bc8f66e3.js"><link rel="prefetch" href="/vigour/assets/js/5.547ab64b.js"><link rel="prefetch" href="/vigour/assets/js/50.33ccbf09.js"><link rel="prefetch" href="/vigour/assets/js/51.a4537e69.js"><link rel="prefetch" href="/vigour/assets/js/52.742a00ea.js"><link rel="prefetch" href="/vigour/assets/js/53.a8c66093.js"><link rel="prefetch" href="/vigour/assets/js/54.e3c75d6b.js"><link rel="prefetch" href="/vigour/assets/js/55.b83e30e7.js"><link rel="prefetch" href="/vigour/assets/js/56.b1896ed4.js"><link rel="prefetch" href="/vigour/assets/js/57.e044df75.js"><link rel="prefetch" href="/vigour/assets/js/58.691b2714.js"><link rel="prefetch" href="/vigour/assets/js/59.437a636d.js"><link rel="prefetch" href="/vigour/assets/js/6.6c2c654a.js"><link rel="prefetch" href="/vigour/assets/js/60.ec5c3723.js"><link rel="prefetch" href="/vigour/assets/js/61.021203ed.js"><link rel="prefetch" href="/vigour/assets/js/62.b36f4a57.js"><link rel="prefetch" href="/vigour/assets/js/63.c05450cf.js"><link rel="prefetch" href="/vigour/assets/js/64.88958d78.js"><link rel="prefetch" href="/vigour/assets/js/65.6bf36464.js"><link rel="prefetch" href="/vigour/assets/js/66.41f1bdba.js"><link rel="prefetch" href="/vigour/assets/js/67.37251a04.js"><link rel="prefetch" href="/vigour/assets/js/68.ba725d65.js"><link rel="prefetch" href="/vigour/assets/js/69.74690479.js"><link rel="prefetch" href="/vigour/assets/js/70.b1a00b85.js"><link rel="prefetch" href="/vigour/assets/js/71.4dd8eace.js"><link rel="prefetch" href="/vigour/assets/js/72.543bdb2e.js"><link rel="prefetch" href="/vigour/assets/js/73.85f3e194.js"><link rel="prefetch" href="/vigour/assets/js/74.d08f12ef.js"><link rel="prefetch" href="/vigour/assets/js/75.9fca9282.js"><link rel="prefetch" href="/vigour/assets/js/76.68180b0b.js"><link rel="prefetch" href="/vigour/assets/js/77.2547e45b.js"><link rel="prefetch" href="/vigour/assets/js/78.9d6ea6a3.js"><link rel="prefetch" href="/vigour/assets/js/79.b4c2005f.js"><link rel="prefetch" href="/vigour/assets/js/80.efe5c402.js"><link rel="prefetch" href="/vigour/assets/js/81.cfdd14fd.js"><link rel="prefetch" href="/vigour/assets/js/82.37156a57.js"><link rel="prefetch" href="/vigour/assets/js/83.519a20ba.js"><link rel="prefetch" href="/vigour/assets/js/84.341659e3.js"><link rel="prefetch" href="/vigour/assets/js/85.2f5a1a71.js"><link rel="prefetch" href="/vigour/assets/js/86.e58cd9f5.js"><link rel="prefetch" href="/vigour/assets/js/87.a4b436a2.js"><link rel="prefetch" href="/vigour/assets/js/88.11646299.js"><link rel="prefetch" href="/vigour/assets/js/89.92d13be2.js"><link rel="prefetch" href="/vigour/assets/js/9.9ceff734.js"><link rel="prefetch" href="/vigour/assets/js/90.d9981d7d.js"><link rel="prefetch" href="/vigour/assets/js/92.7b228fd9.js">
    <link rel="stylesheet" href="/vigour/assets/css/0.styles.83d47a9e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/vigour/" class="home-link router-link-active"><!----> <span class="site-name">Vigour</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="https://github.com/mokunshao/vigour" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="https://github.com/mokunshao/vigour" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/vigour/" class="sidebar-link">Getting Started</a></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Components</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/vigour/components/button.html" class="sidebar-link">Button 按钮</a></li><li><a href="/vigour/components/icon.html" class="sidebar-link">Icon 图标</a></li><li><a href="/vigour/components/input.html" class="sidebar-link">Input 输入框</a></li><li><a href="/vigour/components/tag-input.html" class="sidebar-link">Tag Input</a></li><li><a href="/vigour/components/grid.html" class="sidebar-link">Grid 网格</a></li><li><a href="/vigour/components/layout.html" class="sidebar-link">Layout 布局</a></li><li><a href="/vigour/components/checkbox.html" class="sidebar-link">Checkbox 复选框</a></li><li><a href="/vigour/components/radio.html" class="sidebar-link">Radio 单选框</a></li><li><a href="/vigour/components/toast.html" class="sidebar-link">Toast</a></li><li><a href="/vigour/components/tabs.html" class="sidebar-link">Tabs 标签页</a></li><li><a href="/vigour/components/popover.html" class="sidebar-link">Popover</a></li><li><a href="/vigour/components/collapse.html" class="sidebar-link">Collapse 折叠</a></li><li><a href="/vigour/components/toggle.html" class="sidebar-link">Toggle 切换</a></li><li><a href="/vigour/components/dialog.html" class="sidebar-link">Dialog 对话框</a></li><li><a href="/vigour/components/notification.html" class="sidebar-link">Notification 通知</a></li><li><a href="/vigour/components/cascader.html" class="sidebar-link">Cascader</a></li><li><a href="/vigour/components/dropdown.html" class="sidebar-link">Dropdown 下拉框</a></li><li><a href="/vigour/components/formatted-input.html" class="sidebar-link">Formatted Input</a></li><li><a href="/vigour/components/rating.html" class="sidebar-link">Rating 评分</a></li><li><a href="/vigour/components/tree.html" class="active sidebar-link">Tree</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vigour/components/tree.html#example" class="sidebar-link">Example</a></li><li class="sidebar-sub-header"><a href="/vigour/components/tree.html#props" class="sidebar-link">Props</a></li></ul></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="tree"><a href="#tree" class="header-anchor">#</a> Tree</h1> <div class="language-javascript extra-class"><pre class="language-javascript"><code><span class="token keyword">import</span> <span class="token punctuation">{</span> tree <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vigour'</span><span class="token punctuation">;</span>
<span class="token comment">// tree.name === 'vigour-tree'</span>
</code></pre></div><h2 id="example"><a href="#example" class="header-anchor">#</a> Example</h2> <h3 id="basic-usage"><a href="#basic-usage" class="header-anchor">#</a> Basic usage</h3> <div><ul class="vigour-tree" data-v-bcee4fb4><li class="vigour-tree-item" data-v-70d42422 data-v-bcee4fb4><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon vigour-tree-item-icon-rotate" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>家用电器</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon vigour-tree-item-icon-rotate" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电视</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>超薄电视</span></div> <!----></li><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>全面屏电视</span></div> <!----></li></ul></li><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>空调</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>中央空调</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>变频空调</span></div> <!----></li></ul></li></ul></li><li class="vigour-tree-item" data-v-70d42422 data-v-bcee4fb4><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑整机</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>笔记本</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>游戏本</span></div> <!----></li></ul></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑配件</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>显示器</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>CPU</span></div> <!----></li></ul></li></ul></li></ul></div> <div class="language-html extra-class"><pre class="language-html"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>vigour-tree</span> <span class="token attr-name">:data</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>json<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>vigour-tree</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><div class="language-javascript extra-class"><pre class="language-javascript"><code><span class="token punctuation">[</span>
    <span class="token punctuation">{</span>
      label<span class="token punctuation">:</span> <span class="token string">'家用电器'</span><span class="token punctuation">,</span>
      isExpanded<span class="token punctuation">:</span> <span class="token boolean">true</span><span class="token punctuation">,</span>
      id<span class="token punctuation">:</span> <span class="token number">1</span><span class="token punctuation">,</span>
      children<span class="token punctuation">:</span> <span class="token punctuation">[</span>
        <span class="token punctuation">{</span>
          label<span class="token punctuation">:</span> <span class="token string">'电视'</span><span class="token punctuation">,</span>
          id<span class="token punctuation">:</span> <span class="token number">11</span><span class="token punctuation">,</span>
          isExpanded<span class="token punctuation">:</span> <span class="token boolean">true</span><span class="token punctuation">,</span>
        <span class="token punctuation">}</span><span class="token punctuation">,</span>
      <span class="token punctuation">]</span><span class="token punctuation">,</span>
    <span class="token punctuation">}</span><span class="token punctuation">,</span>
<span class="token punctuation">]</span><span class="token punctuation">;</span>
</code></pre></div><h3 id="ontoggle-onclick"><a href="#ontoggle-onclick" class="header-anchor">#</a> onToggle &amp; onClick</h3> <p>The <code>onToggle</code> and <code>onClick</code> attributes accept a function as an argument.</p> <p><code>onToggle</code> 和 <code>onClick</code> 属性接受一个函数作为参数。</p> <div><ul class="vigour-tree" data-v-bcee4fb4><li class="vigour-tree-item" data-v-70d42422 data-v-bcee4fb4><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon vigour-tree-item-icon-rotate" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>家用电器</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon vigour-tree-item-icon-rotate" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电视</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>超薄电视</span></div> <!----></li><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>全面屏电视</span></div> <!----></li></ul></li><li class="vigour-tree-item" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>空调</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>中央空调</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>变频空调</span></div> <!----></li></ul></li></ul></li><li class="vigour-tree-item" data-v-70d42422 data-v-bcee4fb4><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑整机</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>笔记本</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>游戏本</span></div> <!----></li></ul></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><svg aria-hidden="true" class="vigour-icon vigour-tree-item-icon" data-v-dd3c7748 data-v-70d42422><use xlink:href="#icon-right" data-v-dd3c7748></use></svg> <span class="vigour-tree-item-label" data-v-70d42422>电脑配件</span></div> <ul class="vigour-tree" data-v-70d42422><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>显示器</span></div> <!----></li><li class="vigour-tree-item" style="display:none;" data-v-70d42422 data-v-70d42422><div class="vigour-tree-item-row" data-v-70d42422><!----> <span class="vigour-tree-item-label" data-v-70d42422>CPU</span></div> <!----></li></ul></li></ul></li></ul></div> <div class="language-html extra-class"><pre class="language-html"><code> <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>vigour-tree</span>
  <span class="token attr-name">:data</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>json<span class="token punctuation">&quot;</span></span>
  <span class="token attr-name">:onToggle</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>onToggle<span class="token punctuation">&quot;</span></span>
  <span class="token attr-name">:onClick</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">&quot;</span>onClick<span class="token punctuation">&quot;</span></span>
<span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>vigour-tree</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><div class="language-javascript extra-class"><pre class="language-javascript"><code>methods<span class="token punctuation">:</span> <span class="token punctuation">{</span>
  <span class="token function">onToggle</span><span class="token punctuation">(</span><span class="token parameter">data</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
    console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'onToggle'</span><span class="token punctuation">,</span> data<span class="token punctuation">)</span><span class="token punctuation">;</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
  <span class="token function">onClick</span><span class="token punctuation">(</span><span class="token parameter">data</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
    console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'onClick'</span><span class="token punctuation">,</span> data<span class="token punctuation">)</span><span class="token punctuation">;</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
<span class="token punctuation">}</span><span class="token punctuation">,</span>
</code></pre></div><h2 id="props"><a href="#props" class="header-anchor">#</a> Props</h2> <table><thead><tr><th>prop</th> <th>type</th> <th>default value</th> <th>available value</th></tr></thead> <tbody><tr><td>data</td> <td><code>array</code></td> <td>-</td> <td>-</td></tr> <tr><td>onToggle</td> <td><code>function</code></td> <td>-</td> <td>-</td></tr> <tr><td>onClick</td> <td><code>function</code></td> <td>-</td> <td>-</td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/vigour/components/rating.html" class="prev">Rating 评分</a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/vigour/assets/js/app.8d4caa4f.js" defer></script><script src="/vigour/assets/js/8.b83c43f8.js" defer></script><script src="/vigour/assets/js/91.5a3e06e9.js" defer></script><script src="/vigour/assets/js/22.1269dd55.js" defer></script><script src="/vigour/assets/js/23.b2be9338.js" defer></script>
  </body>
</html>
