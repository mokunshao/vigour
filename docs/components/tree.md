---
title: Tree
---

# Tree

```javascript
import { tree } from 'vigour';
// tree.name === 'vigour-tree'
```

## Example

### Basic usage

<tree-example-1></tree-example-1>

```html
<vigour-tree :data="json"></vigour-tree>
```

```javascript
[
    {
      label: '家用电器',
      isExpanded: true,
      id: 1,
      children: [
        {
          label: '电视',
          id: 11,
          isExpanded: true,
        },
      ],
    },
];
```

### onToggle & onClick

The `onToggle` and `onClick` attributes accept a function as an argument.

`onToggle` 和 `onClick` 属性接受一个函数作为参数。

<tree-example-2></tree-example-2>

```html
 <vigour-tree
  :data="json"
  :onToggle="onToggle"
  :onClick="onClick"
></vigour-tree>
```

```javascript
methods: {
  onToggle(data) {
    console.log('onToggle', data);
  },
  onClick(data) {
    console.log('onClick', data);
  },
},
```
## Props

| prop | type | default value | available value|
| -----| ---- | ---- | --- | --- |
| data | `array` | - | - |
| onToggle | `function` | - | - |
| onClick | `function` | - | - |
