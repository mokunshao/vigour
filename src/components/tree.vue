<template>
  <ul class="vigour-tree">
    <vigour-tree-item
      v-for="(item, index) in data"
      :key="item.id ? item.id : index"
      :data="item"
    ></vigour-tree-item>
  </ul>
</template>

<script>
import TreeItem from './tree-item.vue';

export default {
  name: 'vigour-tree',
  components: {
    [TreeItem.name]: TreeItem,
  },
  props: {
    data: {},
  },
};
</script>

<style lang="scss" scoped>
.vigour-tree {
  border: 1px solid black;
  padding: 1em;
}
</style>
