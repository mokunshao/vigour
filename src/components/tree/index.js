import Tree from './tree.vue';

export { Tree };

export default Tree;
