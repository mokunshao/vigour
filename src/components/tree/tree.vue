<template>
  <ul class="vigour-tree">
    <vigour-tree-item
      v-for="(item, index) in data"
      :key="item.id ? item.id : index"
      :item="item"
    ></vigour-tree-item>
  </ul>
</template>

<script>
import TreeItem from './tree-item.vue';

export default {
  name: 'vigour-tree',
  provide() {
    return {
      tree: {
        onClick: this.onClick,
        onToggle: this.onToggle,
        state: this,
      },
    };
  },
  components: {
    [TreeItem.name]: TreeItem,
  },
  props: {
    data: {
      type: Array,
    },
    onToggle: {
      type: Function,
    },
    onClick: {
      type: Function,
    },
  },
  data() {
    return {
      selected: null,
    };
  },
};
</script>

<style lang="scss" scoped>
.vigour-tree {
  padding: 0;
  margin: 0;
  margin-left: -1em;
}
</style>
