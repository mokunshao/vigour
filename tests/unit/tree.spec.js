import { mount } from '@vue/test-utils';
import { Tree } from '../../src/components/tree/index';

const json = [
  {
    label: '家用电器',
    isExpanded: true,
    id: 1,
    children: [
      {
        label: '电视',
        id: 11,
        isExpanded: true,
      },
    ],
  },
];

describe('Tree', () => {
  it('Tree 存在', () => {
    expect(Tree).toBeTruthy();
  });
  test('renders correctly', () => {
    const wrapper = mount(Tree, {
      propsData: {
        data: json,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
