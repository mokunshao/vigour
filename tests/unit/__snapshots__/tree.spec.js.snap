// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree renders correctly 1`] = `
<ul
  class="vigour-tree"
>
  <li
    class="vigour-tree-item"
  >
    <div
      class="vigour-tree-item-row"
    >
      <svg
        aria-hidden="true"
        class="vigour-icon vigour-tree-item-icon vigour-tree-item-icon-rotate"
      >
        <use
          xlink:href="#icon-right"
        />
      </svg>
       
      <span
        class="vigour-tree-item-label"
      >
        家用电器
      </span>
    </div>
     
    <ul
      class="vigour-tree"
    >
      <li
        class="vigour-tree-item"
      >
        <div
          class="vigour-tree-item-row"
        >
          <!---->
           
          <span
            class="vigour-tree-item-label"
          >
            电视
          </span>
        </div>
         
        <!---->
      </li>
    </ul>
  </li>
</ul>
`;
